### PR TITLE
Merge of terminal_arch_refactor

### PR DESCRIPTION
**Pull request by sergey-chernikov**
Just a conflict resolve between terminal_arch_refactor and futures_dev.

### CODE REVIEW

File: BlocksettleNetworkingLib/Message/ThreadedAdapter.h [hunk 1]
```diff
@@ -43,8 +43,7 @@ namespace bs {
 
       private:
          std::thread processingThread_;
-
-         std::atomic_bool                       continueExecution_ { true };
+         std::shared_ptr<std::atomic_bool>      continueExecution_;
```
Comment (Ation) [inline BlocksettleNetworkingLib/Message/ThreadedAdapter.h:46]: Why it was changed to a pointer?

Comment (sergey-chernikov) [reply BlocksettleNetworkingLib/Message/ThreadedAdapter.h:46]: To persist in a different thread that is created in ThreadedAdapter.

File: BlocksettleNetworkingLib/ApplicationSettings.cpp [hunk 2]
```diff
@@ -95,12 +88,17 @@ namespace {
...
+}
+
 ApplicationSettings::ApplicationSettings(const QString &appName
    , const QString& rootDir)
-   : settings_(QSettings::IniFormat, QSettings::UserScope, SettingsCompanyName, appName)
+   : settings_(QSettings::IniFormat, QSettings::UserScope, appDirName, appName)
 {
    if (rootDir.isEmpty()) {
-      commonRoot_ = AppendToWritableDir(QLatin1String(".."));
+      commonRoot_ = AppendToWritableDir(QLatin1Literal(".."));
```
Comment (Ation) [inline BlocksettleNetworkingLib/ApplicationSettings.cpp:101]: Isn't it the same? Maybe you need QStringLiteral?

**Comment by sergey-chernikov** [comment]
It’s the same now, but should render to literal object in future Qt versions. We need Latin1 literal here.

File: BlocksettleNetworkingLib/ApplicationSettings.cpp (hunk 2 above)
Comment (sergey-chernikov) [reply BlocksettleNetworkingLib/ApplicationSettings.cpp:101]: It’s the same now, but should render to literal object in future Qt versions. We need Latin1 literal here.

File: BlocksettleNetworkingLib/CacheFile.cpp [hunk 3]
```diff
@@ -190,14 +179,14 @@ void CacheFile::purge()
 
 BinaryData CacheFile::get(const BinaryData &key) const
 {
-   QReadLocker lockMap(&rwLock_);
+   std::unique_lock<std::mutex> lock(rwMutex_);
```
Comment (Ation) [inline BlocksettleNetworkingLib/CacheFile.cpp:182]: Please use shared_lock for read lock.

**Review by Ation — APPROVED** [review]

**Pull request merged**